src/regmod/variable.py

```diff
@@ -18,15 +18,75 @@
 
 @dataclass
 class Variable:
+    """Variable class is in charge of storing information of variable including
+    name and priors, and accessing data in the data frame. Name correspondes to
+    column name in the data frame and priors are used to compute the likelihood.
+
+    Parameters
+    ----------
+    name : str
+        Name of the variable corresponding to the column name in the data frame.
+    priors : List[Prior], optional
+        A list of priors for the variable. Default is an empty list.
+
+    Attributes
+    ----------
+    size
+    name : str
+        Name of the variable corresponding to the column name in the data frame.
+    priors : List[Prior]
+        A list of priors for the variable.
+    gprior : GaussianPrior
+        Direct Gaussian prior in `priors`.
+    uprior : UniformPrior
+        Direct Uniform prior in `priors`.
```

Comment:
I'm curious about the design. Why do we want both `priors` and `gprior`/`uprior` as arguments here? According to the function `process_priors`, if the `gprior` is specified, but not included in `priors`, it will give errors and won't be caught (`self.priors.remove(self.gprior)`)

Reply:
Here I want to maintain a list of priors so that users do not need to specify which prior they input. But I agree, it makes the code unnecessarily complicated. We could discuss how to simplify this pair :)